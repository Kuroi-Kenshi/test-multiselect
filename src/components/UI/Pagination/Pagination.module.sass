.pageList
  display: flex
  list-style: none
  padding: 0
  .pageListItem
    margin-right: 5px
.leftArrow
  transform: rotate(180deg)
.pagination
  display: flex
  justify-content: space-between
.btn
  background-color: transparent
  border: none
  cursor: pointer
  text-align: left
