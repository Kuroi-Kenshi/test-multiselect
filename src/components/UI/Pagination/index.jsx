import React from 'react';
import { eventGenerator } from '../../../utils/eventGenerator';

import arrow from '../../../assets/icons/arrow-right.svg'
import s from './Pagination.module.sass';

const Pagination = ({ itemsPerPage, totalItems, pagiante, currentPage }) => {
  const pageNumbers = [];

  for (let i = 1; i <= Math.ceil(totalItems / itemsPerPage); i++) {
    pageNumbers.push(i);
  }

  const togglePage = (number) => {
    if (pageNumbers.length >= number && number > 0) {
      pagiante(number)
    }
  }

  return (
    <div className={s.pagination} data-event={eventGenerator("Click on pagination form", '')}>
      <button className={s.btn} onClick={() => togglePage(currentPage - 1)}>
        <img src={arrow} alt="" className={s.leftArrow} data-event={eventGenerator("Prev page", currentPage - 1)} />
      </button>
      <div data-event={eventGenerator("Click on pagination page number", '')}>{currentPage} из {pageNumbers.length}</div>
      <button className={s.btn} onClick={() => togglePage(currentPage + 1)} >
        <img src={arrow} alt="" data-event={eventGenerator("Next page", currentPage + 1)} />
      </button>
    </div>
  );
};

export default Pagination;