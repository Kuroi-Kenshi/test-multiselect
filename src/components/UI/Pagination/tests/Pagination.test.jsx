import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Pagination from '../index';

const setUp = (props) => shallow(<Pagination {...props} />);

describe("should render Pagination component", () => {

  test("matches", () => {
    const wrapper = setUp({
      itemsPerPage: 10,
      totalItems: 50,
      pagiante: () => { },
      currentPage: 1
    });
    expect(toJson(wrapper)).toMatchSnapshot();
  })
})