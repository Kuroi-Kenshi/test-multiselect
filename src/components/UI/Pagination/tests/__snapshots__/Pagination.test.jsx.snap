// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Pagination component matches 1`] = `
<div
  className="pagination"
  data-event="{ \\"id\\": \\"1636955733074\\", \\"eventName\\": \\"Click on pagination form\\", \\"eventValue\\": \\"\\"}"
>
  <button
    className="btn"
    onClick={[Function]}
  >
    <img
      alt=""
      className="leftArrow"
      data-event="{ \\"id\\": \\"1636955733074\\", \\"eventName\\": \\"Prev page\\", \\"eventValue\\": \\"0\\"}"
      src="test-file-stub"
    />
  </button>
  <div
    data-event="{ \\"id\\": \\"1636955733074\\", \\"eventName\\": \\"Click on pagination page number\\", \\"eventValue\\": \\"\\"}"
  >
    1
     из 
    5
  </div>
  <button
    className="btn"
    onClick={[Function]}
  >
    <img
      alt=""
      data-event="{ \\"id\\": \\"1636955733074\\", \\"eventName\\": \\"Next page\\", \\"eventValue\\": \\"2\\"}"
      src="test-file-stub"
    />
  </button>
</div>
`;
